receiver/prometheusremotewritereceiver/receiver.go

```diff
@@ -39,8 +42,9 @@ type prometheusRemoteWriteReceiver struct {
 	settings     receiver.Settings
 	nextConsumer consumer.Metrics
 
-	config *Config
-	server *http.Server
+	jobInstanceCache map[string]pmetric.ResourceMetrics
```

Comment:
consider using an int64 as the key for this cache, and constructing an identifier using xxhash, similar to what prometheus translation does: https://github.com/prometheus/prometheus/blob/0f0deb77a2f15383914ad60a3309281adb7e8b27/storage/remote/otlptranslator/prometheusremotewrite/helper.go#L94

You can just append the job, a separator, and then the instance.